drivers/src/persistent.rs

```diff
@@ -102,35 +119,47 @@ impl IdevIdCsr {
     }
 }
 
-const _: () = assert!(size_of::<IdevIdCsr>() < memory_layout::IDEVID_CSR_SIZE as usize);
+const _: () = assert!(size_of::<PersistentData>() == memory_layout::PERSISTENT_DATA_SIZE as usize);
+const _: () = assert!(size_of::<ImageManifest>() <= MAN1_SIZE as usize);
+const _: () = assert!(size_of::<ImageManifest>() <= MAN2_SIZE as usize);
+const _: () = assert!(size_of::<FirmwareHandoffTable>() <= FHT_SIZE as usize);
+const _: () = assert!(size_of::<PcrLogArray>() <= PCR_LOG_SIZE as usize);
+const _: () = assert!(size_of::<StashMeasurementArray>() <= MEASUREMENT_LOG_SIZE as usize);
+const _: () = assert!(size_of::<FuseLogArray>() <= FUSE_LOG_SIZE as usize);
+#[cfg(feature = "runtime")]
+const _: () = assert!(size_of::<PcrResetCounter>() <= PCR_RESET_COUNTER_SIZE as usize);
+#[cfg(feature = "runtime")]
+const _: () = assert!(
+    size_of::<AuthManifestImageMetadataCollection>() <= AUTH_MAN_IMAGE_METADATA_MAX_SIZE as usize
+);
+const _: () = assert!(size_of::<IdevIdCsr>() <= IDEVID_CSR_SIZE as usize);
```

Comment:
Are these asserts necessary? I would expect the assert on line 122 to be sufficient. I think we mostly just care that weird alignment hasn't changed the overall size of the struct.

Reply:
There is padding between certain sections / reserved memory for different compilation targets, so if a struct grows beyond the hard coded sizes weird things might happen.

I think it's a good sanity check in case some of these types grow in the future, but could be a bit redundant.

What's your preference?

Reply:
Ok, makes sense. Let's go ahead and leave this.